# PhageTermini Analysis

## Phage DNA Packaging Mechanisms

During the later stages of infection, dsDNA phages generally form **concatenates**, which are cleaved by the **terminase** enzyme during packaging to create mature chromosomes.

There are **four main mechanisms** by which phages recognize and package their DNA:
![image](https://github.com/user-attachments/assets/423ceaa0-1236-4c4f-b546-d8690a661dfc)
1. **Cos-site Recognition**  
   The terminase enzyme introduces staggered cuts at specific cos sites, creating cohesive DNA ends with either 5′ or 3′ overhangs (e.g., **Lambda** and **HK975** phages).

2. **Terminal Repeats (DTRs)**  
   The terminase recognizes fixed positions on the phage genome where direct terminal repeats are generated by extending synthesis at the 3′ ends of staggered nicks. The length of these repeats can vary from just over 100 bases to more than 10,000 bases (e.g., **T36**, **T77**, **T58**, **Spo19**).  
   - Phage **N4** has different terminal repeats on its left and right genome ends.

3. **Pac-site Recognition**  
   The terminase initiates packaging at a specific **pac site** on the phage concatemer, with cuts made at variable positions. This results in circularly permutated genomes with redundant ends that recombine after injection into the host cell (e.g., **P111**, **P2212**).

4. **Headful Packaging (T4-like Phages)**  
   In T4-like phages, there is no specific pac site. Packaging starts randomly and is typically accompanied by host DNA degradation, ensuring that only viral DNA is packaged.




## Computational Practical

### Step 1: Create a directory for your analysis and cd into it
```
mkdir PhageTermini_analysis
cd PhageTermini_analysis
```
### Step 2: Download PhageTerm software
```
wget https://gitlab.pasteur.fr/vlegrand/ptv/-/archive/py3_release_1_light/ptv-py3_release_1_light.tar.gz
```
### Step 2: Extract the dowloaded software file
```
tar -xvf ptv-py3_release_1_light.tar.gz
```
### Step 3: Create conda environment from the `.yml` file
```
conda create -f ptv-py3_release_1_light/PhageTerm_env_py3.yml
```
### Step 4: Activate the conda environemnt
```
conda activate PhageTerm_env_py3
```

### Download test datasets
```
wget test_data.tar https://github.com/ckigenk/Phage-Bioinformatics-Training/raw/refs/heads/main/PhageTerm/test_data.tar
```
### Extract the file
```
tar -xvf test_data.tar
```
### Activate phageterm
```
conda activate PhageTerm_env_py3
```
### Run phageterm
```
ptv-py3_release_1_light/PhageTerm.py -f test_data/COS-3.fastq -r test_data/COS-3.fasta --report_title cohesive -c 20
```
